updaters.py

```diff
@@ -0,0 +1,74 @@
+""" Updater for regular items. """
+from abc import ABCMeta, abstractmethod
+
+
+class AbstractUpdater(ABCMeta):
+
+    @abstractmethod
+    def tick(item):
+        pass
+
+
+class SulfurasUpdater(AbstractUpdater):
+
+    @staticmethod
+    def tick(item):
+        return None
+
+
+class RegularUpdater(AbstractUpdater):
+
+    @staticmethod
+    def tick(item):
+        coeff = max(min(item.sell_in, 1), 0)
+        item.quality -= (coeff * 1 + (1 - coeff) * 2)
+        # if item.sell_in <= 0:
+        #     item.quality -= 2
+        # else:
+        #     item.quality -= 1
+        item.quality = max(0, item.quality)
+        item.sell_in -= 1
+
+
+class AgedBrieUpdater(AbstractUpdater):
+
+    @staticmethod
+    def tick(item):
+        coeff = max(min(item.sell_in, 1), 0)
```

Comment:
This is overcomplicated. Giving up on readability and understandability for a price of three lines less isn't a good idea. 
It's nice to see that you know how to do it in different, kind of interesting way tho.

Reply:
Since it was my suggestion to try and not use IFs I have to give it a try in making it more readable.
```
def new_quality(current_quality, date_expired):
    return current_quality - (date_expired * 1 + (1 - date_expired) * 2)
      
not_expired = max(min(item.sell_in, 1), 0)
item.quality = new_quality(item.quality, not_expired)
```